lib/validator.js

```diff
@@ -25,6 +25,26 @@ class Validator {
         return date instanceof Date
       }
     })
+
+    this.ajv.addKeyword({
+      keyword: 'isSet',
+      type: 'object',
+      errors: false,
+      validate: (type, data, schema) => {
+        if (data instanceof Set) {
+          data = Array.from(data)
+        }
+
+        if (schema.isSet) {
+          // undo what we do in convertSchemaToAjvFormat
+          // to prevent infinite recursive call
+          delete schema.isSet
+          schema.type.pop('object')
```

Comment:
array's pop method always returns the last item and doesn't have any arguments

Reply:
In 7d169eaabe1685e0919e832018bf93b120625f6a! But also found that the `JSON.stringify` inside the error message  doesn't handle `Set` well. Can we please have another PR for that?